app/channel.go

```diff
@@ -757,15 +789,37 @@ func (a *App) RestoreChannel(channel *model.Channel, userID string) (*model.Chan
 			},
 		}
 
-		if _, err := a.CreatePost(post, channel, false, true); err != nil {
+		if _, err := a.CreatePost(c, post, channel, false, true); err != nil {
 			mlog.Warn("Failed to post unarchive message", mlog.Err(err))
 		}
+	} else {
+		a.Srv().Go(func() {
+			systemBot, err := a.GetSystemBot()
+			if err != nil {
+				mlog.Error("Failed to post unarchive message", mlog.Err(err))
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.